pkg/virt-api/rest/authorizer.go

```diff
@@ -115,13 +116,25 @@ func (a *authorizor) getUserName(header http.Header) (string, error) {
 	return "", fmt.Errorf("a valid user header is required for authorization")
 }
 
+func hasPrefixIgnoreCase(s, prefix string) bool {
+	return len(s) >= len(prefix) && strings.EqualFold(s[:len(prefix)], prefix)
+}
+
+func unescapeExtraKey(encodedKey string) string {
+	key, err := url.PathUnescape(encodedKey) // Decode %-encoded bytes.
+	if err != nil {
+		return encodedKey // Always record extra strings, even if malformed/unencoded.
+	}
+	return key
+}
+
 func (a *authorizor) getUserExtras(header http.Header) map[string]authv1.ExtraValue {
 	extras := map[string]authv1.ExtraValue{}
 
 	for _, prefix := range a.userExtraHeaderPrefixes {
 		for k, v := range header {
-			if strings.HasPrefix(k, prefix) {
-				extraKey := strings.TrimPrefix(k, prefix)
+			if hasPrefixIgnoreCase(k, prefix) {
+				extraKey := unescapeExtraKey(strings.ToLower(k[len(prefix):]))
```

Comment:
Changes "inspired" by: https://github.com/kubernetes/apiserver/blob/master/pkg/authentication/request/headerrequest/requestheader.go#L180-L204

Reply:
@mhenriks I understand the unescape logic, i don't understand the implications of reducing the key to lower case though. Why is that done here?

Reply:
@davidvossel not exactly sure why, probably subjectaccessreview API expects lower case? Nevertheless I think it is best to do exactly what the k8s reference apiserver does

Reply:
Maybe this explains it:  https://github.com/kubernetes/apiserver/blob/c8097e3f30c749fa023c708d54ba855cbd53fbc6/pkg/authentication/user/user.go#L39-L41